src/signTxCatalystRegistration.c

```diff
@@ -0,0 +1,576 @@
+	}
+
+	// TODO - is it worth declaring a policy for this?
+	subctx->ui_step = HANDLE_NONCE_STEP_DISPLAY;
+	signTxCatalystRegistration_handleNonce_ui_runStep();
+}
+
+// ============================== CONFIRM ==============================
+
+enum {
+	HANDLE_CONFIRM_STEP_FINAL_CONFIRM,
+	HANDLE_CONFIRM_STEP_DISPLAY_HASH,
+	HANDLE_CONFIRM_STEP_RESPOND,
+	HANDLE_CONFIRM_STEP_INVALID,
+};
+
+static void signTxCatalystRegistration_handleConfirm_ui_runStep()
+{
+	TRACE("UI step %d", subctx->ui_step);
+	TRACE_STACK_USAGE();
+	ui_callback_fn_t* this_fn = signTxCatalystRegistration_handleConfirm_ui_runStep;
+
+	UI_STEP_BEGIN(subctx->ui_step);
+	UI_STEP(HANDLE_CONFIRM_STEP_FINAL_CONFIRM) {
+		ui_displayPrompt(
+		        "Confirm voting key",
+		        "registration?",
+		        this_fn,
+		        respond_with_user_reject
+		);
+	}
+	UI_STEP(HANDLE_CONFIRM_STEP_DISPLAY_HASH) {
+		ui_displayHexBufferScreen(
+		        "Auxiliary data hash",
+		        (uint8_t*) &ctx->auxDataHash,
+		        SIZEOF(ctx->auxDataHash),
+		        this_fn
+		);
+	}
+	UI_STEP(HANDLE_CONFIRM_STEP_RESPOND) {
+		struct {
+			uint8_t auxDataHash[AUX_DATA_HASH_LENGTH];
+			uint8_t signature[ED25519_SIGNATURE_LENGTH];
+		} wireResponse;
+
+		ASSERT(SIZEOF(ctx->auxDataHash) == AUX_DATA_HASH_LENGTH);
+		os_memmove(wireResponse.auxDataHash, ctx->auxDataHash, AUX_DATA_HASH_LENGTH);
+
+		ASSERT(SIZEOF(subctx->stateData.registrationSignature) == ED25519_SIGNATURE_LENGTH);
+		os_memmove(wireResponse.signature, subctx->stateData.registrationSignature, ED25519_SIGNATURE_LENGTH);
+
+		io_send_buf(SUCCESS, (uint8_t*) &wireResponse, SIZEOF(wireResponse));
+		advanceState();
+	}
+	UI_STEP_END(HANDLE_CONFIRM_STEP_INVALID);
+}
+
+__noinline_due_to_stack__
+static void signTxCatalystRegistration_handleConfirmAPDU(uint8_t* wireDataBuffer MARK_UNUSED, size_t wireDataSize)
+{
+	{
+		//sanity checks
+		CHECK_STATE(STATE_CATALYST_REGISTRATION_CONFIRM);
+
+		ASSERT(wireDataSize < BUFFER_SIZE_PARANOIA);
+	}
+
+	{
+		explicit_bzero(&subctx->stateData, SIZEOF(subctx->stateData));
+	}
+
+	{
+		// no data to receive
+		VALIDATE(wireDataSize == 0, ERR_INVALID_DATA);
+	}
+
+	{
+		{
+			uint8_t votingPayloadHashBuffer[CATALYST_REGISTRATION_PAYLOAD_HASH_LENGTH];
+			auxDataHashBuilder_catalystRegistration_finalizePayload(auxDataHashBuilder, votingPayloadHashBuffer, AUX_DATA_HASH_LENGTH);
+			getCatalystVotingRegistrationSignature(
+			        &subctx->stakingKeyPath,
+			        votingPayloadHashBuffer, CATALYST_REGISTRATION_PAYLOAD_HASH_LENGTH,
+			        (uint8_t*) &subctx->stateData.registrationSignature, ED25519_SIGNATURE_LENGTH
+			);
+		}
+		auxDataHashBuilder_catalystRegistration_addSignature(auxDataHashBuilder, (uint8_t*) &subctx->stateData.registrationSignature, ED25519_SIGNATURE_LENGTH);
+		auxDataHashBuilder_catalystRegistration_addAuxiliaryScripts(auxDataHashBuilder);
+
+		auxDataHashBuilder_finalize(auxDataHashBuilder, (uint8_t*) &ctx->auxDataHash, AUX_DATA_HASH_LENGTH);
```

Comment:
We do the `(uint8_t*)` quite often here and it looks suspicious. AFAIK we should use
```ctx->auxDataHash```
instead of 
```(uint8_t*) &ctx->auxDataHash```, probably everywhere.

Reply:
Unrelated to this, github does not show the right place at this point to me, but there is a compiler warning:
```
src/auxDataHashBuilder.c:20:1: warning: 'static' ignored on this declaration [-Wmissing-declarations]
static enum {
```

Reply:
Another unrelated:
```
	txHashBuilder_assertCanLeaveAuxData(builder);

	// add fee item into the main tx body map
```
the comment should not be about fee.